src/natcap/invest/urban_nature_access.py

```diff
@@ -347,328 +504,647 @@ def execute(args):
+                'greenspace_sup_dem_paths_by_pop_group':
+                    greenspace_supply_demand_by_group_paths,
+                'proportional_pop_paths_by_pop_group':
+                    proportional_population_paths,
+                'undersupply_by_pop_group': supply_population_paths['under'],
+                'oversupply_by_pop_group': supply_population_paths['over'],
             },
-            task_name='Rasterize the admin units vector',
-            target_path_list=[file_registry['aois_ids']],
+            task_name=(
+                'Aggregate supply-demand to admin units (by pop groups)'),
+            target_path_list=[file_registry['aois']],
             dependent_task_list=[
-                aoi_reprojection_task, lulc_alignment_task]
-        )
+                aoi_reprojection_task,
+                *greenspace_supply_demand_by_group_tasks,
+                *proportional_population_tasks.values(),
+                *supply_population_tasks['under'].values(),
+                *supply_population_tasks['over'].values(),
+            ])
 
-        for pop_field in split_population_fields:
-            field_value_map = _read_field_from_vector(
-                file_registry['reprojected_aois'],
-                ID_FIELDNAME, pop_field)
-            for supply_type, supply_filepath in (
-                    ('under', file_registry['undersupplied_population']),
-                    ('over', file_registry['oversupplied_population'])):
-                groupname = re.sub(POP_FIELD_REGEX, '', pop_field)
-                supply_to_group_filepath = os.path.join(
-                    intermediate_dir,
-                    f'{supply_type}supplied_population_{groupname}{suffix}.tif'
-                )
-                _ = graph.add_task(
-                    _reclassify_and_multiply,
+    # Greenspace budget, supply/demand and over/undersupply rasters are the
+    # same for uniform radius and for split greenspace modes.
+    if args['search_radius_mode'] in (RADIUS_OPT_UNIFORM,
+                                      RADIUS_OPT_GREENSPACE):
+        # This is "SUP_DEMi_cap" from the user's guide
+        per_capita_greenspace_budget_task = graph.add_task(
+            pygeoprocessing.raster_calculator,
+            kwargs={
+                'base_raster_path_band_const_list': [
+                    (file_registry['greenspace_supply'], 1),
+                    (float(args['greenspace_demand']), 'raw')
+                ],
+                'local_op': _greenspace_budget_op,
+                'target_raster_path': file_registry['greenspace_budget'],
+                'datatype_target': gdal.GDT_Float32,
+                'nodata_target': FLOAT32_NODATA
+            },
+            task_name='Calculate per-capita greenspace budget',
+            target_path_list=[file_registry['greenspace_budget']],
+            dependent_task_list=[
+                greenspace_supply_task,
+            ])
+
+        # This is "SUP_DEMi" from the user's guide
```

Comment:
I just wanted to check if these comments still hold up with what's being worked into the Users Guide.

Reply:
Yep!  But they're terrible names.  I just found out about the paper that represents this model and they use some different names, so I'll probably update everything to use those in the near future.